packages/components/src/TabBar/TabBar.component.js

```diff
@@ -116,8 +116,9 @@ function TabBar(props) {
 			))}
 		</Tab.Content>
 	);
-	if (responsive && showDropdown) {
-		const selectedItem = items[selectedKey - 1];
+
+	const selectedItem = items.find(item => item.key === selectedKey);
```

Comment:
Does this mean that when no tab selected, it will not render responsive layout?
I propose when no tab selected, render responsive layout with first tab item instead.